src/query.rs

```diff
@@ -43,7 +46,9 @@ impl Query {
     ) -> Result<ShoppingCartItem> {
         let db_client = ctx.data_unchecked::<Database>();
         let collection: Collection<User> = db_client.collection::<User>("users");
-        query_shoppingcart_item(&collection, id).await
+        let user = query_shoppingcart_item_user(&collection, id).await?;
+        authenticate_user(&ctx, user._id)?;
```

Comment:
why authentication logic in the entity resolver?

Reply:
Removed now